src/main/java/com/google/api/codegen/transformer/py/PythonImportSectionTransformer.java

```diff
@@ -0,0 +1,173 @@
+    ImportSectionView.Builder importSection = ImportSectionView.newBuilder();
+    importSection.appImports(generateInitCodeAppImports(context, specItemNodes));
+    return importSection.build();
+  }
+
+  private List<ImportFileView> generateInitCodeAppImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ImmutableList.Builder<ImportFileView> appImports = ImmutableList.builder();
+    appImports.add(generateApiImport(context));
+    appImports.addAll(generateProtoImports(context, specItemNodes));
+    appImports.addAll(generatePageStreamingImports(context));
+    return appImports.build();
+  }
+
+  private ImportFileView generateApiImport(MethodTransformerContext context) {
+    return createImport(
+        context.getApiConfig().getPackageName(),
+        context.getNamer().getApiWrapperVariableName(context.getInterface()));
+  }
+
+  private List<ImportFileView> generateProtoImports(
+      MethodTransformerContext context, Iterable<InitCodeNode> specItemNodes) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    Set<ImportFileView> protoImports = new TreeSet<>(importFileViewComparator());
+    for (InitCodeNode item : specItemNodes) {
+      TypeRef type = item.getType();
+      if (!type.isRepeated() && type.isMessage()) {
+        String fullName = typeTable.getFullNameFor(type);
+        String nickname = typeTable.getNicknameFor(type);
+        protoImports.add(generateAppImport(fullName, nickname));
+      } else if (type.isEnum()) {
+        protoImports.add(createImport(context.getApiConfig().getPackageName(), "enums"));
+      }
+    }
+    return new ArrayList<>(protoImports);
+  }
+
+  private List<ImportFileView> generatePageStreamingImports(MethodTransformerContext context) {
+    ImmutableList.Builder<ImportFileView> pageStreamingImports = ImmutableList.builder();
+    if (context.getMethodConfig().isPageStreaming()) {
+      ImportTypeView.Builder callOptionsImport = ImportTypeView.newBuilder();
+      callOptionsImport.fullName("CallOptions");
+      callOptionsImport.nickname("");
+
+      ImportTypeView.Builder initialPageImport = ImportTypeView.newBuilder();
+      initialPageImport.fullName("INITIAL_PAGE");
+      initialPageImport.nickname("");
+
+      ImportFileView.Builder fileImport = ImportFileView.newBuilder();
+      fileImport.moduleName("google.gax");
+      fileImport.types(ImmutableList.of(callOptionsImport.build(), initialPageImport.build()));
+      pageStreamingImports.add(fileImport.build());
+    }
+    return pageStreamingImports.build();
+  }
+
+  private Comparator<ImportFileView> importFileViewComparator() {
+    return new Comparator<ImportFileView>() {
+      @Override
+      public int compare(ImportFileView o1, ImportFileView o2) {
+        if (!o1.moduleName().equals(o2.moduleName())) {
+          return o1.moduleName().compareTo(o2.moduleName());
+        }
+
+        if (o1.types().size() != o2.types().size()) {
```

Comment:
This logic seems odd to me: does this mean that if an API adds a few types to a proto file, all of the imports in the GAPIC might get shuffled around? When is it not possible to compare based on the file name alone?